TestModels/Documentation/Model/documentation.smithy

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+namespace simple.documentation
+
+/// A service that supports the operation of getting things.
+@aws.polymorph#localService(
+  sdkId: "SimpleDocumentation",
+  config: SimpleDocumentationConfig,
+)
+service SimpleDocumentation {
+  version: "2021-11-01",
+  resources: [ Widget ],
+  operations: [ GetThing ],
+  errors: [ CouldntGetTheThingError ],
+}
+
+/// Fancy configuration things to make getting things even thingier.
+structure SimpleDocumentationConfig {}
+
+/// A thing that you can get from the service.
+structure Thing {
+  /// The name of the thing.
+  name: String
+}
+
+/// Call this to get a thing.
+operation GetThing {
+  input: GetThingInput
+  output: GetThingOutput
+}
+
+/// Inputs for getting a thing.
+structure GetThingInput {
+  /// The name of the thing to get, obviously.
+  @required
+  name: String
+}
+
+/// Outputs for getting a thing.
+structure GetThingOutput {
+  /// The thing that you just got.
+  @required
+  thing: Thing
+}
+
+/// A different kind of thing you can get.
+resource Widget {
+  operations: [ SetWidgetName ]
+}
+
+@aws.polymorph#reference(resource: Widget)
+structure WidgetReference {}
+
+operation SetWidgetName {
+  input: SetWidgetNameInput
+}
+
+structure SetWidgetNameInput {
+  @required
+  name: String
+}
+
+/// Error returned when we couldn't get the thing.
+@error("server")
+structure CouldntGetTheThingError {
+  /// Error message (this documentation is ignored!)
```

Comment:
REALLY! This is ignored!
Yeah... fuck this `///` syntax.

Reply:
It would happen with `@documentation` as well, the issue is OUR code generation is special casing `message` on errors and ignoring it.

I'll be more specific in the comment.

Reply:
OH. OK. 
If this was any other shape, it work, right?
For example:
```
@javadoc("Configures Key Store's KMS Key ARN restrictions.")
union KMSConfiguration {
  @documentation("<blah>")
  // The Key Store (released on 2023-07-24) only allowed for this.
  kmsKeyArn: com.amazonaws.kms#KeyIdType,

  ///<more-blah>
  kmsMRKeyArn: com.amazonaws.kms#KeyIdType,
}
```
<blah> and <more-blah> would be detected, correct?

Because they are right now with JavaDoc.
Evidence:
- [Smithy Model](https://github.com/aws/aws-cryptographic-material-providers-library/blob/633e4c1639fd3fb5725779943547b751b647d6a7/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/KeyStore.smithy#L108-L125)
- [Smithy Generated Java](https://github.com/aws/aws-cryptographic-material-providers-library/blob/633e4c1639fd3fb5725779943547b751b647d6a7/AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystore/model/KMSConfiguration.java#L8-L31)

Reply:
Ah definitely. And the test model has quite a few cases of documentation on members.